sql: TestAggregatesMonitorMemory failed

sql.TestAggregatesMonitorMemory [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6139981?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6139981?buildTab=artifacts#/) on master @ [5c2c62ecf1bea60c807edc6b4da22d900ad4ae03](https://github.com/cockroachdb/cockroach/commits/5c2c62ecf1bea60c807edc6b4da22d900ad4ae03):


```
=== RUN   TestAggregatesMonitorMemory
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/233e6aa31ff7ad29c896f0479cb5d0c4/logTestAggregatesMonitorMemory3477543293
    test_log_scope.go:80: use -show-logs to present logs inline
    builtin_mem_usage_test.go:85: pq: read-users: root: memory budget exceeded: 10240 bytes requested, 1044480 currently allocated, 1048576 bytes in budget
    panic.go:500: -- test log scope end --
--- FAIL: TestAggregatesMonitorMemory (4.24s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAggregatesMonitorMemory.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18723

This test sometimes seems to fail with the error here, but much more often it panics with
```
panic: MoveToDraining called in state StateDraining with err: root: memory budget exceeded: 10240 bytes requested, 1044480 currently allocated, 1048576 bytes in budget
```

I've bisected both the failure, I think, to https://github.com/cockroachdb/cockroach/pull/84761.
I'm thinking perhaps there's something to learn here:
1. Why do we panic on some memory limit error
2. Why did the memory usage/accounting increase in that PR. 

I see you're bumping the memory limit; I wanna make sure we're not missing something deeper.

--- a/pkg/sql/builtin_mem_usage_test.go
+++ b/pkg/sql/builtin_mem_usage_test.go
@@ -24,22 +24,22 @@ import (
 	"github.com/lib/pq"
 )
 
-// lowMemoryBudget is the memory budget used to test builtins are recording
-// their memory use. The budget needs to be large enough to establish the
-// initial database connection, but small enough to overflow easily. It's set
-// to be comfortably large enough that the server can start up with a bit of
-// extra space to overflow.
-const lowMemoryBudget = 1 << 20 /* 1MiB */
-
 // rowSize is the length of the string present in each row of the table created
 // by createTableWithLongStrings.
-const rowSize = 30000
+const rowSize = 50000
 
 // numRows is the number of rows to insert in createTableWithLongStrings.
 // numRows and rowSize were picked arbitrarily but so that rowSize * numRows >
 // lowMemoryBudget, so that aggregating them all in a CONCAT_AGG or
 // ARRAY_AGG will exhaust lowMemoryBudget.
-const numRows = 50
+const numRows = 100
+
+// lowMemoryBudget is the memory budget used to test builtins are recording
+// their memory use. The budget needs to be large enough to establish the
+// initial database connection, but small enough to overflow easily. It's set
+// to be comfortably large enough that the server can start up with a bit of
+// extra space to overflow.
+const lowMemoryBudget = rowSize*numRows - 1
 
 // createTableWithLongStrings creates a table with a modest number of long strings,
 // with the intention of using them to exhaust a memory budget.

